pkg/cli/cli.go

```diff
@@ -134,12 +165,20 @@ func WithExtraCommands(cmds ...*cobra.Command) Option {
 
 // initialize initializes the cli.
 func (c *cli) initialize() error {
+	// Initialize cli with globally-relevant flags or flags that determine
+	// certain plugin type's configuration.
+	if err := c.parseBaseFlags(); err != nil {
```

Comment:
why do we need this globally?  isn't this just needed on the `init` invocation?

Reply:
The CLI needs to pre-parse the `--plugins` global flag for each command, so I figured wrapping all flag pre-parsing in one method made sense. The `cli.doGenericHelp` flag is set by `cli.parseBaseFlags` so `init` can still print generic help if `--project-version` is not set.

Reply:
> The CLI needs to pre-parse the --plugins global flag for each command,

wait, does it?  I thought we figured out that it didn't...

Reply:
Summary of discussion: having a global `--plugins` flag is not in scope for phase 1 since the type of plugin used in `init` solely determines what plugin is used for subsequent commands (the `layout` config key). A global `--plugins` flag will be discussed in phase 2: #1378.